netcreateorg/netcreate-2018
# Feature: Edge Weight

We've implemented a first pass at showing edge weights.

Now when you add an edge, the line width of the link will change based on the number of links between the two nodes. 


# TO TEST

1. Check out `dev-bl/edge-weight`
2. `npm run dev`
3. Create a new edge.
4. Note the size (stroke/line width) of the edge.  It should be about 1 pixel wide.
5. Create a second edge between the the same two nodes.
6. The width of the edge should now be twice as large.
7. As you add more edges, the line width should keep increasing.
8.  If you delete an edge, the width should decrease.

# NOTES/CAVEATS

* The line width is actually rendered as the square of the number of links between the two nodes.  This makes the difference between one and two nodes more obvious (e.g. if it's one link, the width is 1 pixel, but if there are two links, the the size is 2^2=4. This is necessary because the difference between 1 pixel and 2 pixels is fairly subtle and hard to catch.

* The system is actually drawing two edges: one from each node to the other.  They are invisible because they are drawn on top of each other.  This might become problematic if we decide to draw the lines with some alpha transparency.

* Edge weights are currently only calculated when a new edge is added.  So if you open up an existing database, the edge weights will display incorrect data.  If you edit the edge and re-save it, the system should then properly recalculate.  We can add a routine later that can pre-processes all nodes.

* For debugging purposes, you can set `DBG=true` in EdgeTable to view edge ids and the size parameter in the expanded table. This is helpful for verifying if the size calculations are correct.  Set it back to `false` to hide the columns again.

* There was a bug in EdgeTable where clicking on the "Edit" button would result in a javascript error because edge ids were not strings  This should be fixed now.
